src/unix/socket.c

```diff
@@ -482,19 +538,17 @@ static sysreturn unixsock_connect(struct sock *sock, struct sockaddr *addr,
     }
 
     struct sockaddr_un *unixaddr = (struct sockaddr_un *) addr;
-    tuple t;
-    buffer b;
     unixsock listener, peer;
-    if (filesystem_get_tuple(unixaddr->sun_path, &t) < 0) {
-        return -ECONNREFUSED;
-    }
-    b = get(t, sym(socket)); // XXX untyped binary
-    if (!b || (buffer_length(b) != sizeof(u64))) {
-        return -ECONNREFUSED;
-    }
-    listener = pointer_from_u64(*((u64 *) buffer_ref(b, 0)));
-    assert(listener);
+    int rv = lookup_socket(&listener, unixaddr->sun_path);
+    if (rv != 0)
+        return rv;
     if (!s->connecting) {
+        if (s->sock.type & SOCK_DGRAM) {
+            if (!(listener->sock.type & SOCK_DGRAM))
```

Comment:
Here and in the previous line, `&` should be replaced by `==`

Reply:
Yup, I got these mixed up with the flags argument.